src/components/tables/DataTable.vue

```diff
@@ -51,7 +51,7 @@ export default {
   },
   methods: {
     initTable: function() {
-      $.fn.dataTable.moment(moment.localeData().longDateFormat('ll'));
+      $.fn.dataTable.moment(moment().creationData().locale._longDateFormat.LL, moment.locale());
```

Comment:
what does this output?

Reply:
@YaroslavOvdii It doesn't look like you need to change the structure of the call here.

```suggestion
      $.fn.dataTable.moment(moment.localeData().longDateFormat('LL'));
```

or

```suggestion
      $.fn.dataTable.moment(moment.localeData().longDateFormat('LL'), moment.locale());
```

...should do. `moment.localeData().longDateFormat(x)` is the shortform and official way to get locale specific format. See https://momentjs.com/docs/#/i18n/locale-data/

And also please explain your solution as @squallstar suggested.

Reply:
@squallstar this output was changed to match the date format that we use in the data [table](https://github.com/Fliplet/fliplet-widget-organization-dashboard/blob/master/src/components/tables/DataTableCell.vue#L53), added in this [PR](https://github.com/Fliplet/fliplet-widget-organization-dashboard/pull/32). And as we've added localization to the date format I've passing a user local to the function that determinates date columns correctly.

Reply:
Is this because you're attempting to sort results by the displayed date? If so, that would not meet the requirement.

Results must be sorted by their ISODATE as returned from the API, e.g. `2020-12-31T00:00:00`, not by the formatted day with the user's locale.

Reply:
@squallstar Yes, you are right. We are trying to sort by displayed date. We are sorting by displayed date because we are putting only transformed date into the data table lib.
If we need to sort them by ISODATE, that is means we need to pass it to the lib. And remade this plugin so it would look for the correct value to sort.
Should I create a new PR with that solution or add a new commit to this PR?

Reply:
@YaroslavOvdii have you verified whether the plugin supports sorting a column by its raw value rather than the displayed value?

Reply:
@squallstar we can do it like [this plugin](https://github.com/Fliplet/fliplet-widget-organization-dashboard/blob/master/vendor/dataTables.numString.js). As you may see we put sorting value into the data attribute and we can do whatever we want with that value.

What I suggest to do:
1. At the step when we generate a cell, we put not transformed Date to the data attribute.
2. When the data table library will be initialized in the `unshift` method we will grand cell a type for sorting according to the data attribute we put earlier.
3. When a user will try to sort cells will be sorted by their type. And we will determine what sorting value it will have in this method `order['moment-' + format + '-pre']`.

Will it be okay to remade our date plugin by that example?

Reply:
@squallstar Sorry for not answering fully to your question.
The plugin will work with data over what we had initialized the data table library. But now we initializing it over a rendered table. Which is rendered with VUE. That is why we are getting a string of the dom elements that we put into a cell.
In the past, we have a [discussion ](https://flipletoutsourced.slack.com/archives/CLPSXU4RF/p1593076032001800)about rendering tables by passing data array into the library but we wouldn't get any advantages from that way and increase code read difficulty.

Reply:
@YaroslavOvdii I'm good with your suggested implementation. Just to confirm, the requirements are:

1. Date sorting must work by ISODATE A-Z (because that's the only acceptable result), not by actual rendered output (which may differ between locales)
2. Updating vendor libraries must be avoided as much as possible. Any update must be commented so it's clear that we have made a change, e.g.

```js
/* Updated by Fliplet on 2020-12-14 to <description of the update> */
code
```